core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -421,12 +424,16 @@ namespace kagome::consensus::babe {
   void BabeImpl::processSlot() {
     BOOST_ASSERT(keypair_ != nullptr);
 
-    if (lottery_->getEpoch() != current_epoch_) {
-      auto epoch_res = block_tree_->getEpochDescriptor(
-          current_epoch_.epoch_number, best_block_.hash);
-      BOOST_ASSERT(epoch_res.has_value());
-      const auto &epoch = epoch_res.value();
+    auto epoch_res = block_tree_->getEpochDescriptor(
+        current_epoch_.epoch_number, best_block_.hash);
+    BOOST_ASSERT(epoch_res);
+    const auto &epoch = epoch_res.value();
+    auto authority_index_res =
+        getAuthorityIndex(epoch.authorities, keypair_->public_key);
+    BOOST_ASSERT_MSG(authority_index_res.has_value(), "Authority is not known");
```

Comment:
I think, failure might be happened here in case iff authority list is changed. It is enough to just return from here.